### PR TITLE
Improved build process and consistency

### PR DESCRIPTION
**Pull request by davidz627**
Improved build to only rebuild when source files changed. Changed output directory to bin/ for consistency with external-attacher and driver-registrar.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -12,7 +12,7 @@ COPY ./examples/flex-debug.sh /opt/csi-provisioner/
 
 # install the go kube piece of provisioner
 
-COPY ./_output/csi-provisioner /opt/csi-provisioner/
+COPY ./bin/csi-provisioner /opt/csi-provisioner/
```
Comment (rootfs) [inline Dockerfile]: let's make _output/bin for binaries. Change here and in Makefile and README too.

Comment (jsafrane) [reply Dockerfile]: It should be the same as attacher, which uses `bin/` too.

Comment (childsb) [reply Dockerfile]: I do not like bin/ for build artifacts.... It's a common directory for scripts or other real binaries that might be part of the project..  _output/bin would be fine.

**Comment by davidz627** [comment]
Closing due to inactivity and staleness.

**Pull request closed without merging**